rt/emul/mini/src/main/resources/org/apidesign/vm4brwsr/emul/lang/java_lang_Number.js

```diff
@@ -1,572 +1,2671 @@
+  sr_Statics$: 1,
+  O: 1
+});
+$c_sr_Statics$.prototype.$classData = $d_sr_Statics$;
+var $n_sr_Statics$ = (void 0);
+function $m_sr_Statics$() {
+  if ((!$n_sr_Statics$)) {
+    $n_sr_Statics$ = new $c_sr_Statics$().init___()
+  };
+  return $n_sr_Statics$
+}
+/** @constructor */
+function $c_jl_Number() {
+  $c_O.call(this)
+}
+$c_jl_Number.prototype = new $h_O();
+$c_jl_Number.prototype.constructor = $c_jl_Number;
+/** @constructor */
+function $h_jl_Number() {
+  /*<skip>*/
+}
+$h_jl_Number.prototype = $c_jl_Number.prototype;
+/** @constructor */
+function $c_jl_Throwable() {
+  $c_O.call(this);
+  this.s$1 = null;
+  this.e$1 = null;
+  this.stackTrace$1 = null
+}
+$c_jl_Throwable.prototype = new $h_O();
+$c_jl_Throwable.prototype.constructor = $c_jl_Throwable;
+/** @constructor */
+function $h_jl_Throwable() {
+  /*<skip>*/
+}
+$h_jl_Throwable.prototype = $c_jl_Throwable.prototype;
+$c_jl_Throwable.prototype.fillInStackTrace__jl_Throwable = (function() {
+  var v = $g.Error.captureStackTrace;
+  if ((v === (void 0))) {
+    try {
+      var e$1 = {}.undef()
+    } catch (e) {
+      var e$2 = $m_sjsr_package$().wrapJavaScriptException__O__jl_Throwable(e);
+      if ((e$2 !== null)) {
+        if ($is_sjs_js_JavaScriptException(e$2)) {
+          var x5 = e$2;
+          var e$3 = x5.exception$4;
+          var e$1 = e$3
+        } else {
+          var e$1;
+          throw $m_sjsr_package$().unwrapJavaScriptException__jl_Throwable__O(e$2)
+        }
+      } else {
+        var e$1;
+        throw e
+      }
+    };
+    this.stackdata = e$1
+  } else {
+    $g.Error.captureStackTrace(this);
+    this.stackdata = this
+  };
+  return this
+});
+$c_jl_Throwable.prototype.getMessage__T = (function() {
+  return this.s$1
+});
+$c_jl_Throwable.prototype.toString__T = (function() {
+  var className = $objectGetClass(this).getName__T();
+  var message = this.getMessage__T();
+  return ((message === null) ? className : ((className + ": ") + message))
+});
+$c_jl_Throwable.prototype.init___T__jl_Throwable = (function(s, e) {
+  this.s$1 = s;
+  this.e$1 = e;
+  this.fillInStackTrace__jl_Throwable();
+  return this
+});
+function $is_jl_Throwable(obj) {
+  return (!(!((obj && obj.$classData) && obj.$classData.ancestors.jl_Throwable)))
+}
+function $isArrayOf_jl_Throwable(obj, depth) {
+  return (!(!(((obj && obj.$classData) && (obj.$classData.arrayDepth === depth)) && obj.$classData.arrayBase.ancestors.jl_Throwable)))
+}
+/** @constructor */
+function $c_s_util_hashing_MurmurHash3$() {
+  $c_s_util_hashing_MurmurHash3.call(this);
+  this.seqSeed$2 = 0;
+  this.mapSeed$2 = 0;
+  this.setSeed$2 = 0
+}
+$c_s_util_hashing_MurmurHash3$.prototype = new $h_s_util_hashing_MurmurHash3();
+$c_s_util_hashing_MurmurHash3$.prototype.constructor = $c_s_util_hashing_MurmurHash3$;
+/** @constructor */
+function $h_s_util_hashing_MurmurHash3$() {
+  /*<skip>*/
+}
+$h_s_util_hashing_MurmurHash3$.prototype = $c_s_util_hashing_MurmurHash3$.pro
```

Comment:
@sjrd, if there is an easy way to avoid these hashing functions to be generated, I happily use that.

Reply:
I'm afraid there isn't. They are transitively needed by the `hashCode` function of `JavaScriptException`, which is used in `Throwable.fillInStackTrace`. It's a bunch of internal stuff on which you don't have control, as a user of Scala.js.

The thing is that Scala.js has never been optimized for "tiny" programs like yours, so the constant cost of an almost empty program is not something we've tried to reduce. It could happen one day.

There is a *hard* way, though, which would be to fork Scala.js for your specific use case, and define a trivial
```scala
override def hashCode(): Int = exception.hashCode()
```
in [`JavaScriptException`](https://github.com/scala-js/scala-js/blob/v1.0.0-M3/library/src/main/scala/scala/scalajs/js/JavaScriptException.scala).

In fact, maybe we should do that anyway in the core ...